elastic-search-filler/src/main/java/in/ramanujam/elasticsearch/filler/ElasticSearchFiller.java

```diff
@@ -99,4 +87,26 @@ public static void writeIsFinished( boolean isFinished ) throws JsonProcessingEx
     client.prepareIndex( index, ElasticSearchProperties.getInstance().getElasticsearchIsFinishedKey(), ElasticSearchProperties.getInstance().getElasticsearchIsFinishedKey() )
             .setSource( mapper.writeValueAsBytes( isFinishedMap ) ).get();
   }
+
+  private JsonStreamDataSupplier<MinerRecord> getStreamDataSupplier()
+  {
+    InputStream inputStream = null;
+    try
+    {
+      inputStream = elasticSearchFile.getInputStream();
+    }
+    catch( IOException e )
+    {
+      e.printStackTrace();
+    }
+
+    return JsonStreamDataSupplier.mapping( MinerRecord.class )
+              .forStream( inputStream )
+              .build();
+  }
+
+  public long elementsLeft( int fromNumber )
+  {
+    return getStreamDataSupplier().stream().count() - fromNumber;
```

Comment:
This causes opening file, parsing, creating stream 2 times more
Use supplier hasNext() to find whether reached end of file